sys/net/routing/rpl/rpl_control_messages.c

```diff
@@ -840,8 +840,17 @@ void rpl_recv_DAO(void)
                       ipv6_addr_to_str(addr_str, IPV6_MAX_ADDR_STR_LEN, &rpl_opt_target_buf->target),
                       ipv6_addr_to_str(addr_str, IPV6_MAX_ADDR_STR_LEN, &ipv6_buf->srcaddr),
                       (rpl_opt_transit_buf->path_lifetime * my_dodag->lifetime_unit));
-                rpl_add_routing_entry(&rpl_opt_target_buf->target, &ipv6_buf->srcaddr,
-                                      rpl_opt_transit_buf->path_lifetime * my_dodag->lifetime_unit);
+
+                /* If we are the target we don't need to add a new FIB entry
+                 * This is a bit hack-ish, but sufficient for now
+                */
+                if ( my_address.uint32[3] != rpl_opt_target_buf->target.uint32[3] ) {
```

Comment:
Hmm, when I enable `DEBUG`, `rpl_opt_target_buf->target` reveals the link local address ~5 lines above.
Maybe there's something wrong with the `rpl_opt_target_buf` construction/access